utils.lua

```diff
@@ -150,4 +150,16 @@ function nn.utils.addSingletonDimension(t, dim)
   return view
 end
 
+function nn.utils.contiguousView(output, input, ...)
+  output = output or input.new()
+  if input:isContiguous() then
+    output:view(input, ...)
+  else
+    output:resizeAs(input)
+    output:copy(input)
+    output:view(...)
```

Comment:
Because `view` is allocated in [here](https://github.com/torch/torch7/blob/master/Tensor.lua#L446), creating a new tensor, and afterwards it's returned [here](https://github.com/torch/torch7/blob/master/Tensor.lua#L465). The code doesn't change `result`, which is what you pass to the function. What you want instead is to call [this](https://github.com/torch/torch7/blob/master/Tensor.lua#L431), because it will effectively change the passed tensor.

Here is a snippet reproducing the problem
```lua
a = torch.rand(3,3)
a:view(-1);
print(a) -- a is still 3x3
```